luigi/worker.py

```diff
@@ -678,6 +707,26 @@ def _log_remote_tasks(self, running_tasks, n_pending_tasks, n_unique_pending):
             if n_unique_pending:
                 logger.debug("There are %i pending tasks unique to this worker", n_unique_pending)
 
+    def _get_work_task_id(self, get_work_response):
+        if get_work_response['task_id'] is not None:
+            return get_work_response['task_id']
+        elif 'batch_id' in get_work_response:
+            task = load_task(
+                module=get_work_response.get('task_module'),
+                task_name=get_work_response['task_family'],
+                params_str=get_work_response['task_params'],
+            )
+            self._scheduler.add_task(
+                worker=self._id,
+                task_id=task.task_id,
+                module=get_work_response.get('task_module'),
+                family=get_work_response['task_family'],
+                params=task.to_str_params(),
+                status=RUNNING,
+                batch_id=get_work_response['batch_id'],
+            )
```

Comment:
We need this because we moved the batching logic into the worker. When the scheduler sends the worker a bunch of batched tasks, it doesn't know what task the worker will end up running. The worker might start running one of the batched tasks or it could create a new task that the scheduler doesn't know about yet. So it has to inform the scheduler how it's running the batch.

We could add a separate API call for this response, but it would likely have to call `add_task` at least as a first step. I guess it could do that and then mark it running after to keep the `status != RUNNING` invariant if that's important.